lib/iaas/ComputeClient.js

```diff
@@ -1,22 +1,34 @@
 'use strict';
 
+const _ = require('lodash');
 const Promise = require('bluebird');
 const AWS = require('aws-sdk');
 const MsRestAzure = require('ms-rest-azure');
 const ComputeManagementClient = require('azure-arm-compute');
 const errors = require('../errors');
+const NotImplemented = errors.NotImplemented;
 
 class ComputeClient {
   static createComputeClient(settings) {
     return Promise
       .try(() => {
         switch (settings.provider) {
         case 'amazon':
-          return new AWS.EC2({
-            accessKeyId: settings.keyId,
-            secretAccessKey: settings.key,
-            region: settings.region
-          });
+          {
+            let ec2_config = {
```

Comment:
I think we should use lowerCamelCase for object (variable) names. 
Suggesting as per: https://github.com/felixge/node-style-guide